app.ts

```diff
@@ -0,0 +1,183 @@
+import { ajax, css } from "jquery";
+let firstNumber: number = 0;
+let lastNumber: number = 0;
+
+
+async function fetchRecordCount(): Promise<number> {
+	try {
+		const recordCount = await fetch(`http://localhost:2050/recordCount`);
+		if (!recordCount.ok) {
+			throw new Error('Failed to fetch record count');
+		}
+		const data = await recordCount.json()
+		return data;
+	} catch (error) {
+		console.error('Error fetching the record count:', error);
+		throw error;
+	}
+}
+
+function fetchColumns(): void {
+	fetch("http://localhost:2050/columns")
+		.then((response: Response) => {
+			return response.json() as Promise<string[]>;
+		})
+		.then((columns: string[]) => {
+			const colArray = columns;
+
+			for (let c = 0; c < colArray.length; c++) {
+				$(".head").append(`<th>${colArray[c]}</th>`);
+			}
+		})
+}
+fetchColumns()
+
+function adjustRowsByScreenHeight() {
+	const screenHeight = window.innerHeight;
+	const availableHeight = screenHeight - 105 // subtracts the space used from the screeen
+	let rowHeight = 35
+	if (availableHeight <= 0) {
+		return 0;
+	} else {
+		let maxRows = Math.floor(availableHeight / rowHeight);
+		return maxRows;
+	}
+}
+
+$(window).on('resize', async () => {
+	let resizeTimeout: number;
+	resizeTimeout = setTimeout(async () => {
+		$('#loader').show()
+		await displayRecords()
+		let inputValue = $('#searchInput').val(); // priotizes the search input value if available
+		if (inputValue !== '') {
+			await updateRecordsAndResize(Number(inputValue))
+		}
+		$('#loader').hide();
+	}, 500);
+})
+
+async function fetchRecords(from: number, to: number): Promise<any[]> {
+	const response = await fetch(`http://localhost:2050/records?from=${from}&to=${to}`);
+	if (!response.ok) {
+		throw new Error("Sorry, there's a problem with the network");
+	}
+	return response.json();
+}
+
+async function displayRecords(): Promise<void> {
+	try {
+		$('#loader').show()
+		let count = await fetchRecordCount() - 1;
```

Comment:
You shouldn't have to make the backend calls the whole time, call once and save then in variables to be reused